corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -518,23 +532,136 @@ hqDefine('app_manager/js/app_manager', [
+                        // Replace spaces with underscore
+                        var term = params.term.replace(/ /g, '_');
+                        return {
+                            id: term,
+                            text: term,
+                        };
+                    },
+                });
+
+                $.ajax({
+                    method: 'GET',
+                    url: initialPageData.reverse('all_case_types'),
+                    success: function (data) {
+                        var existingCaseTypes = data.existing_case_types;
+                        self.deprecatedCaseTypes(data.deprecated_case_types);
+
+                        // Add existing case types as options
+                        existingCaseTypes.forEach(function (caseType) {
+                            var option = new Option(caseType, caseType, false, false);
+                            $caseType.append(option);
+                        });
+                    },
+                });
+
+                $caseType.on('change', function () {
+                    var valueNoSpaces = $(this).val();
+                    var $formGroup = $(this).closest('.form-group');
+                    var $help = $('#new_case_type_help');
+                    var $error = $('#new_case_type_error');
+                    var $createBtn = $('#case-type-create-btn');
+
+                    // Reset error states
+                    $formGroup.removeClass('has-error');
+                    $help.show();
+                    $error.hide();
+                    $createBtn.prop('disabled', false);
+
+                    if (!valueNoSpaces) {
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (module.valueIncludeInvalidCharacters(valueNoSpaces)) {
+                        $formGroup.addClass('has-error');
+                        $error.text(module.invalidCharErrorMsg);
+                        $error.show();
+                        $help.hide();
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (module.valueIsReservedWord(valueNoSpaces)) {
+                        $formGroup.addClass('has-error');
+                        $error.text(module.reservedWordErrorMsg);
+                        $error.show();
+                        $help.hide();
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
+
+                    if (self.deprecatedCaseTypes().includes(valueNoSpaces)) {
+                        $formGroup.addClass('has-error');
+                        $error.text(module.deprecatedCaseTypeErrorMsg);
+                        $error.show();
+                        $help.hide();
+                        $createBtn.prop('disabled', true);
+                        return;
+                    }
```

Comment:
Could we extract the common logic here into something like a `displayError(errorMsg)` function?

Reply:
c2d344a183bf7122301753a199ada18e5cf0f8ff